src/interchange/yaml.zig

```diff
@@ -994,6 +962,74 @@ pub fn Parser(comptime enc: Encoding) type {
             return .init(E.Array, .{ .items = .moveFromList(&seq) }, sequence_start.loc());
         }
 
+        /// Should only be used with expressions created with the YAML parser. It assumes
+        /// only null, boolean, number, string, array, object are possible. It also only
+        /// does pointer comparison with arrays and objects (so exponential merges are avoided)
+        fn yamlMergeKeyExprEql(l: Expr, r: Expr) bool {
+            if (std.meta.activeTag(l.data) != std.meta.activeTag(r.data)) {
+                return false;
+            }
+
+            return switch (l.data) {
+                .e_null => true,
+                .e_boolean => |l_boolean| l_boolean.value == r.data.e_boolean.value,
+                .e_number => |l_number| l_number.value == r.data.e_number.value,
+                .e_string => |l_string| l_string.eql(E.String, r.data.e_string),
+
+                .e_array => |l_array| l_array == r.data.e_array,
+                .e_object => |l_object| l_object == r.data.e_object,
+
+                else => unreachable,
+            };
+        }
+
+        const MappingProps = struct {
+            list: std.array_list.Managed(G.Property),
```

Comment:
It would be nice to make this a private field, and then provide a method to take ownership of the list.

Reply:
good idea

Reply:
made it private and added `pub fn moveList`